websites/api-docs-smoke-test/src/generated-api-specs/test/resources/ByProjectKey.raml

```diff
@@ -1,4 +1,4 @@
-#ByProjectKey
+(resourceName): ByProjectKey
```

Comment:
It works, but I think a custom comment line would be the cleaner way to do this than assuming a specific property being on top although object semantics don't guarantee property order.

Reply:
When you read this with yaml, the property will always be there. And you can directly access it. Accessing the comment may be more tricky. Cause the yaml reader will not put comments in the object.

Reply:
@jenschude  We're filtering the RAML files that have to be read with a low-level  "read the first line from this file" implementation that doesn't even read the full file into memory.  

That is / was necessary to not handle the tons of untyped fragments and includes in the original RAML, parsing them just to decide that it's not a root API definition was too slow. 

That's why the implementation requires literally the first line of a file to be read analogous to shebangs and other "Magic numbers" in the first bytes of a file.

Reply:
So for the generated RAML files I can guarantee that this will be also at the first line. But I could also just duplicate it in a comment. Doesn't matter for me. ;)

Reply:
thanks!  It's more consistent parsing and a better understandable codebase for us if everything is "comment in first line" based